tests/suites/test_suite_x509parse.function

```diff
@@ -1250,6 +1250,86 @@ exit:
 }
 /* END_CASE */
 
+/* BEGIN_CASE depends_on:MBEDTLS_FS_IO:MBEDTLS_X509_CRT_PARSE_C:MBEDTLS_CERTS_C:MBEDTLS_SHA1_C:MBEDTLS_RSA_C */
+void mbedtls_x509_crt_parse_subjectkeyid( char * crt_path, char *subjectKeyId, int subjectKeyIdLength, int retVal )
+{
+    mbedtls_x509_crt crt;
+    int i = 0;
+    int result = 0;
+
+    mbedtls_x509_crt_init( &crt );
+
+    TEST_ASSERT( (mbedtls_x509_crt_parse_file( &crt, crt_path )& 0xFFFF) == retVal );
+
+    if(retVal != 0)
+    {
+        while(i < subjectKeyIdLength)
+        {
+            result |= crt.subject_key_id.p[i] != subjectKeyId[i*2];
+            result |= crt.subject_key_id.p[i+1] != subjectKeyId[i*2+1];
+            i++;
+        }
+
+        TEST_ASSERT(result == 0);
+    }
+
+exit:
+    mbedtls_x509_crt_free( &crt );
+}
+/* END_CASE */
+
+/* BEGIN_CASE depends_on:MBEDTLS_FS_IO:MBEDTLS_X509_CRT_PARSE_C:MBEDTLS_CERTS_C:MBEDTLS_SHA1_C:MBEDTLS_RSA_C */
+void mbedtls_x509_crt_parse_authoritykeyid( char * crt_path, char *authorityKeyId_keyId, int keyIdLength, char *authorityKeyId_issuer, char *authorityKeyId_serial, int serialLength , int retVal)
+{
+    mbedtls_x509_crt crt;
+    int i = 0;
+    int result = 0;
+    int bufferCounter = 0;
+    size_t issuerCounter = 0;
+
+    mbedtls_x509_crt_init( &crt );
+
+    TEST_ASSERT( (mbedtls_x509_crt_parse_file( &crt, crt_path )& 0xFFFF) == retVal );
```

Comment:
The result should not be masked, just check the equality between the expected value and the return value.

Reply:
Corrected, but the solution of one of your previous findings had to be changed:

In test_suite_x509parse.data in testcase:
X509 CRT parse Authority Key Id - Wrong Issuer Tag 4

This:
MBEDTLS_ERR_ASN1_UNEXPECTED_TAG | MBEDTLS_ERR_X509_INVALID_NAME

Had to be changed to this:
MBEDTLS_ERR_ASN1_UNEXPECTED_TAG + MBEDTLS_ERR_X509_INVALID_NAME

Because:
MBEDTLS_ERR_ASN1_UNEXPECTED_TAG  = 0xFF9E
MBEDTLS_ERR_X509_INVALID_NAME        = 0xDC80

So if we OR them the FF will hide DC, and 9E will hide 80, because the bits are already set. Adding them solves the problem